crates/sui/src/sui_commands.rs

```diff
@@ -1516,142 +1498,97 @@ async fn genesis(
-                basic_auth: None,
-                chain_id: None,
-            }),
-            None => {
-                if accept_defaults {
-                    print!(
-                        "Creating config file [{:?}] with default (devnet) Full node server and ed25519 key scheme.",
-                        wallet_conf_path
-                    );
-                } else {
-                    print!(
-                        "Config file [{:?}] doesn't exist, do you want to connect to a Sui Full node server [y/N]?",
-                        wallet_conf_path
-                    );
-                }
-                if accept_defaults
-                    || matches!(read_line(), Ok(line) if line.trim().to_lowercase() == "y")
-                {
-                    let url = if accept_defaults {
-                        String::new()
-                    } else {
-                        print!(
-                            "Sui Full node server URL (Defaults to Sui Testnet if not specified) : "
-                        );
-                        read_line()?
-                    };
-                    Some(if url.trim().is_empty() {
-                        SuiEnv::testnet()
-                    } else {
-                        print!("Environment alias for [{url}] : ");
-                        let alias = read_line()?;
-                        let alias = if alias.trim().is_empty() {
-                            "custom".to_string()
-                        } else {
-                            alias
-                        };
-                        SuiEnv {
-                            alias,
-                            rpc: url,
-                            ws: None,
-                            basic_auth: None,
-                            chain_id: None,
-                        }
-                    })
-                } else {
-                    None
-                }
-            }
-        };
-
-        if let Some(env) = env {
-            let keystore_path = match wallet_conf_path.parent() {
-                // Wallet config was created in the current directory as a relative path.
-                Some(parent) if parent.as_os_str().is_empty() => {
-                    std::env::current_dir().context("Couldn't find current directory")?
-                }
-
-                // Wallet config was given a path with some parent (could be relative or absolute).
-                Some(parent) => parent
-                    .canonicalize()
-                    .context("Could not find sui config directory")?,
+    if wallet_conf_file.exists() {
+        return Ok(());
+    }
 
-                // No parent component and the wallet config was the empty string, use the default
-                // config.
-                None if wallet_conf_path.as_os_str().is_empty() => sui_config_dir()?,
+    // prompt user
+    if !accept_defaults {
+        println!(
+            "No sui config found in `{}`, create one [Y/n]?",
```

Comment:
```suggestion
        print!(
            "No sui config found in `{}`, create one [Y/n]? ",
```
It might be more understandable as a prompt by leaving the carriage return off.